ds3-contract-comparator-print/src/main/java/com/spectralogic/ds3contractcomparator/print/utils/HtmlRowGeneratorUtils.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * ******************************************************************************
+ *   Copyright 2016-2017 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+package com.spectralogic.ds3contractcomparator.print.utils;
+
+import com.google.common.collect.ImmutableList;
+import com.spectralogic.ds3contractcomparator.print.htmlprinter.models.body.rows.Row;
+import org.apache.commons.beanutils.BeanUtils;
+import org.apache.commons.beanutils.PropertyUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Optional;
+
+import static com.spectralogic.ds3autogen.utils.NormalizingContractNamesUtil.removePath;
+
+/**
+ * Utilities for generating the {@link ImmutableList} of {@link Row} which
+ * represent Ds3 Objects using reflection.
+ */
+public final class HtmlRowGeneratorUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HtmlRowGeneratorUtils.class);
+
+    private HtmlRowGeneratorUtils() {
+        //pass
+    }
+
+    /**
+     * Retrieves the specified property of type {@link ImmutableList}
+     */
+    public static <T, N> ImmutableList<N> getListPropertyFromObject(final Field field, final T object) {
+        if (object == null) {
+            return ImmutableList.of();
+        }
+        try {
+            field.setAccessible(true);
+            final Object objectField = field.get(object);
+            if (objectField == null) {
+                return ImmutableList.of();
+            }
+            if (objectField instanceof ImmutableList) {
+                return (ImmutableList<N>) objectField;
+            }
+            throw new IllegalArgumentException("Object should be of type ImmutableList, but was: " + object.getClass().toString());
+        } catch (final IllegalAccessException e) {
+            LOG.warn("Could not retrieve list element {} from object of class {}: {}",
```

Comment:
Make sure the exception is captured in the log message so that way the stack trace is included in the log message